api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -360,6 +371,10 @@ message Bootstrap {
   // Envoy only supports ListenerManager for this field and Envoy Mobile
   // supports ApiListenerManager.
   core.v3.TypedExtensionConfig listener_manager = 37;
+
+  // Optional field to set the application logs format. If this field is set, it will override
+  // the default log format, or the format that is specified by the :option:`--log-format` command line option.
```

Comment:
Looking at the `--service-node` option and it relation to [Node.id](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/api/envoy/config/core/v3/base.proto#L156), it seems that the CLI argument overrides the bootstrap config ([relevant test](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/test/server/server_test.cc#L1026)). 
I suggest to keep it consistent, i.e., CLI overrides Bootstrap overrides default value.

Please also add a test that validates that the correct precedence is kept.

Reply:
Thanks for finding another exmple. However, did you look at the use case, I worte in the application logs doc? If the config doesn't override the CLI option, this won't be available to achieve. I think that it's good enough reason to keep this order of precendance, even at the cost of inconsistantcy with other options precedance..

Reply:
Not sure I follow - why can't we have the bootstrap config be used when `--log-format` is not specified by the command line?

Reply:
I'm not saying that it technically can't be, but the problem is this: since the bootstrap config option is loaded only after some logs are written (the initialization logs) it causes inconsistency, meaning that initialization logs will be printed in default format, and the rest of the logs will be written in the bootstrap configured format.

However, this can be mitigated, if the user configures both ``--log-format`` in manual JSON format, and the bootstrap config. This way, initialization logs will be written by the ``--log-format`` JSON format config, and the rest will be by the bootstrap JSON config.

If we let the CLI option override, then this mitigation is not possible. If you both think that this mitigation isn't important then we can give this up and let the CLI config a precedence. (For me, the initialization logs don't help too much, so I wob't insist on this..). 

Please lmk what you think @adisuissa, @snowp

Reply:
BTW, implementation-wise, it doesn't seem straightforward, because ``options.logFormat()`` gets a default value, so there needs to be a way at the server context to know if this was set in the first place, and this would require some refactoring.

Reply:
I think there will always be cases of "logging" before the log-format (even the one provided using the command line) is set. For example, IIUC [this](https://github.com/envoyproxy/envoy/blob/dc860e850c577484d31b7b7e3697d6de95cc6d01/source/exe/stripped_main_base.cc#L60) can be printed before the log-format is being used.

From a user's experience point-of-view, I still think it makes sense that the command line (if given) overrides whatever is set in the bootstrap (see #2269). The way I would expect things this to work in practice is that there's some bootstrap config that is typically used, and if the operator wants to override it, they add a CLI argument (similar to [overriding the node ID](https://www.envoyproxy.io/docs/envoy/latest/operations/cli.html#cmdoption-config-yaml)). If possible, it may be beneficial to keep this behavior.

That said, I don't feel strongly about this. cc @envoyproxy/envoy-maintainers for other ideas.

Reply:
Got it, I think we can stick to this behavior and let it override